integration-tests/emulated/tests/people/people-kusama/src/tests/governance.rs

```diff
@@ -72,3 +77,454 @@ fn relay_commands_add_registrar() {
+						Transact {
+							origin_kind,
+							require_weight_at_most: Weight::from_parts(500_000_000, 500_000),
+							call: remove_username_authority.encode().into(),
+						}
+					]))),
+				});
+
+			assert_ok!(remove_authority_xcm_msg.dispatch(origin));
+
+			assert_expected_events!(
+				Kusama,
+				vec![
+					RuntimeEvent::XcmPallet(pallet_xcm::Event::Sent { .. }) => {},
+				]
+			);
+		});
+
+		// Final event check.
+		PeopleKusama::execute_with(|| {
+			type RuntimeEvent = <PeopleKusama as Chain>::RuntimeEvent;
+
+			assert_expected_events!(
+				PeopleKusama,
+				vec![
+					RuntimeEvent::Identity(pallet_identity::Event::AuthorityRemoved { .. }) => {},
+					RuntimeEvent::MessageQueue(pallet_message_queue::Event::Processed { success: true, .. }) => {},
+				]
+			);
+		});
+	}
+}
+
+#[test]
+fn relay_commands_add_remove_username_authority_wrong_origin() {
+	let people_kusama_alice = PeopleKusama::account_id_of(ALICE);
+
+	let (origin_kind, origin) = (
+		OriginKind::SovereignAccount,
+		<Kusama as Chain>::RuntimeOrigin::signed(people_kusama_alice.clone()),
+	);
+
+	Kusama::execute_with(|| {
+		type Runtime = <Kusama as Chain>::Runtime;
+		type RuntimeCall = <Kusama as Chain>::RuntimeCall;
+		type RuntimeEvent = <Kusama as Chain>::RuntimeEvent;
+		type PeopleCall = <PeopleKusama as Chain>::RuntimeCall;
+		type PeopleRuntime = <PeopleKusama as Chain>::Runtime;
+
+		let add_username_authority =
+			PeopleCall::Identity(pallet_identity::Call::<PeopleRuntime>::add_username_authority {
+				authority: people_kusama_runtime::MultiAddress::Id(people_kusama_alice.clone()),
+				suffix: b"suffix1".into(),
+				allocation: 10,
+			});
+
+		let add_authority_xcm_msg = RuntimeCall::XcmPallet(pallet_xcm::Call::<Runtime>::send {
+			dest: bx!(VersionedLocation::from(Location::new(0, [Parachain(1004)]))),
+			message: bx!(VersionedXcm::from(Xcm(vec![
+				UnpaidExecution { weight_limit: Unlimited, check_origin: None },
+				Transact {
+					origin_kind,
+					require_weight_at_most: Weight::from_parts(500_000_000, 500_000),
+					call: add_username_authority.encode().into(),
+				}
+			]))),
+		});
+
+		assert_ok!(add_authority_xcm_msg.dispatch(origin));
+
+		assert_expected_events!(
+			Kusama,
+			vec![
+				RuntimeEvent::XcmPallet(pallet_xcm::Event::Sent { .. }) => {},
+			]
+		);
+	});
+
+	// Check events system-parachain-side
+	PeopleKusama::execute_with(|| {
+		type RuntimeEvent = <PeopleKusama as Chain>::RuntimeEvent;
+
+		assert_expected_events!(
+			PeopleKusama,
+			vec![
+				RuntimeEvent::MessageQueue(pallet_message_queue::Event::ProcessingFailed { error: ProcessMessageError::Unsupported, .. }) => {},
+			]
+		);
+	});
+
+	// I mistakenly assumed that to test the removal of an authority would need one to exist.
+	// However, since the origin check is the very first extrinsic in `remove_username_authority`,
+	// an authority need not exist to test the safety of the origin check.
```

Comment:
```suggestion
	// Since the origin check is the very first extrinsic in `remove_username_authority`, an
	// authority need not exist to test the safety of the origin check.
```

Reply:
The dreaded first-person slip. Thanks for noticing.